### PR TITLE
Midi 74/dstart prediction

### PR DESCRIPTION
**Pull request by WojciechMat**
I have created BinsToDstartDataset for holding samples with low-resolution midi data as src and high-resolution tokenized dstart values as tgt. 
My approach was to create a seperate QuantizerWithDstart quantizer, which would append two dstart columns to a quantized sequence instead of one: 'dstart_bin' and 'tgt_dstart_bin'.

Instead of creating seperate function for preprocessing dataset I have added these lines to data.prepare_dataset.process_record:
```python
        if "tgt_dstart_bin" in piece_quantized.df.keys():
            sequence |= {"tgt_dstart_bin": quantized.tgt_dstart_bin.astype("int16").values.T}
```

 It was the fastest approach to implement and it does not require creating seperate tokenizers nor quantizing data again, so it is time-efficient and works fine.

Maybe you have some suggestions as to how to quantize both for src and for tgt in a time-efficient manner, that looks better and/or is simpler to comprehend?

### CODE REVIEW

**Comment by WojciechMat** [comment]
I overfitted 8 batches for 100 epochs and got to ~0.5 loss. 
In case you have been wandering what are the results of overfitted model:

image

I guess it just found out that the majority of keys are played rather closely to each other. I might also try producing linear bins instead of exponential. 
For now I started training the model on whole maestro dataset.

File: utils.py [hunk 1]
```diff
@@ -60,23 +71,77 @@ def load_cached_dataset(cfg: DictConfig, split="test") -> BinsToVelocityDataset:
...
+            hf_dataset = load_dataset(cfg.dataset_name, split=split)
+
+            args = [hf_dataset, n_dstart_bins, n_velocity_bins, n_duration_bins, cfg.sequence_size]
+            try:
+                if cfg.dataset_class == "BinsToDstartDataset":
+                    args.append(cfg.n_tgt_dstart_bins)
+            finally:
+                pass
+
+            dataset = eval(cfg.dataset_class)(*args)
```
Comment (roszcz) [inline utils.py]:
```suggestion
```

This logic feels too convoluted and too deep into the dataset loading process. I think the decision between different dataset classes should be the first thing to happen:


```python
def load_cached_dataset(
        cfg: DictConfig,
        split: str = "test",
) -> TokenizedMidiDataset:
    if cfg.prediction_target == "velocity":
        dataset = load_cached_velocity_dataset(cfg=cfg, split=split)
    if cfg.prediction_target == "dstart":
        dataset = load_cached_dstart_dataset(cfg=cfg, split=split)
    
    return dataset
```

Comment (WojciechMat) [reply utils.py]: sure, will fix that later

**Comment by WojciechMat** [comment]
A model with n=3 layers in encoder and decoder, after 5 epochs of training reaches loss of 2.19 on validation.
image

The results seem to make a lot more sense than in overfitted model, but listening to pieces with dstart predicted by model (100 bins) is often rather painful.
image

**Comment by WojciechMat** [comment]
last commit is needed to run current version of training, but I still need to change the way I go about dataset class selection
